test/agentchat/contrib/test_web_surfer.py

```diff
@@ -0,0 +1,168 @@
+import os
+import sys
+import re
+import pytest
+from autogen import ConversableAgent, UserProxyAgent, config_list_from_json
+from autogen.oai.openai_utils import filter_config
+from conftest import skip_openai
+
+sys.path.append(os.path.join(os.path.dirname(__file__), ".."))
+from test_assistant_agent import KEY_LOC, OAI_CONFIG_LIST  # noqa: E402
+
+BLOG_POST_URL = "https://microsoft.github.io/autogen/blog/2023/04/21/LLM-tuning-math"
+BLOG_POST_TITLE = "Does Model and Inference Parameter Matter in LLM Applications? - A Case Study for MATH | AutoGen"
+BING_QUERY = "Microsoft"
+
+try:
+    from autogen.agentchat.contrib.web_surfer import WebSurferAgent
+except ImportError:
+    skip_all = True
+else:
+    skip_all = False
+
+try:
+    from openai import OpenAI
+except ImportError:
+    skip_openai = True  # noqa: F811, set skip_openai to True since it's not installed.
+
+try:
+    BING_API_KEY = os.environ["BING_API_KEY"]
```

Comment:
Not sure if it's a good idea to put keys together in OAI_CONFIG_LIST instead?

Reply:
I don't think we can put the Big keys in OAI_CONFIG_LIST because that is so intimately tied to the OpenAI ChatCompletion API. We literally use **kwargs to pass the config straight to the call. Adding Bing keys there would necessitate config_list filtering in places where we currently don't need filtering. The OAI_CONFIG_LIST's singular purpose is for LLMs.